pkg/syslog-ng-ctl/stats_prometheus.go

```diff
@@ -154,6 +154,39 @@ func transformEventDelayMetric(delayMetric *io_prometheus_client.MetricFamily, d
 	delayMetric.Type = io_prometheus_client.MetricType_GAUGE.Enum()
 }
 
+// Workaround for a bug in older syslog-ng/AxoSyslog versions where the output of STATS PROMETHEUS was overescaped.
+// Escapes \ as \\ everywhere except for the allowed sequences: \\, \n, \"
+func sanitizeBuggyFormat(output string) string {
+	var fixedOutput strings.Builder
+
+	length := len(output)
+	for i := 0; i < length; i++ {
+		c := output[i]
+
+		if c != '\\' {
+			fixedOutput.WriteByte(c)
+			continue
+		}
+
+		if i+1 >= length {
+			fixedOutput.WriteString("\\\\")
```

Comment:
could write these without escaping using backticks, but I'm not sure if that's less or more confusing 🤷
```suggestion
			fixedOutput.WriteString(`\\`)
```

Reply:
I think this looks slightly better :)